### ISSUE TITLE
Route group callable

### ISSUE THREAD

**Issue by AydinHassan** [slimphp/Slim#1976]
Hi,

I was trying to organise a route group definition into an invokable class to tidy up some files and noticed that I didn't have access to app, so I couldn't define any child routes.

It seems `$app` is only binded to closures. Is there any reason we don't just pass `$app` as a parameter to the route group callable at: https://github.com/slimphp/Slim/blob/3.x/Slim/RouteGroup.php#L45 ?

If not, I would be happy to create a PR :)

**Comment by geggleto** [comment]
the app class gets bound to the group.

``` php
$app->group('/admin', function () {
    //We are now in the context of $app
    $this->get('/users', ''); //Example, is the same as $app->get
});
```

**Comment by AydinHassan** [comment]
Only if it is a closure: https://github.com/slimphp/Slim/blob/3.x/Slim/RouteGroup.php#L41-L43. I was trying to use an invokable class.

**Comment by geggleto** [comment]
oh well that's interesting....

Using DI methods you can inject the router into your class which would allow you to register routes, but I guess it wouldn't be nearly as nice.

**Comment by danopz** [comment]
This should be resolved by #2229
